src/sqlfluff/dialects/dialect_mysql.py

```diff
@@ -2721,6 +2722,37 @@ class ReturnStatementSegment(BaseSegment):
     )
 
 
+class SetTransactionStatementSegment(BaseSegment):
+    """A `SET TRANSACTION` statement.
+
+    As specified in https://dev.mysql.com/doc/refman/8.0/en/set-transaction.html
+    """
+
+    type = "set_transaction_statement"
+    match_grammar: Matchable = Sequence(
+        "SET",
+        OneOf("GLOBAL", "SESSION", optional=True),
+        "TRANSACTION",
+        Delimited(
+            AnySetOf(
```

Comment:
Chaining Delimited and AnySetOf doesn't really work, as Delimited inherits from OneOf - can you remove the AnySetOf here?

Reply:
My intention was to only allow a max of 2 of these options, as you can't have more than 2 of these options at once. e.g., this won't work:
```sql
SET TRANSACTION ISOLATION LEVEL SERIALIZABLE, READ ONLY, READ WRITE;
```
I didn't see any args in `Delimiter()` to allow a max # of items.
Is there a better way to do this?

Reply:
Nah, there's no way to get what you're going for here cleanly. We'd need some kind of DelimitedSet operator, which we don't have at the mo